core-bundle/src/DependencyInjection/Configuration.php

```diff
@@ -422,38 +423,10 @@ static function (array $uris): bool {
      */
     private function canonicalize(string $value): string
     {
-        $resolved = [];
-        $chunks = preg_split('#([\\\\/]+)#', $value, -1, PREG_SPLIT_DELIM_CAPTURE | PREG_SPLIT_NO_EMPTY);
+        $windowsPath = false !== strpos($value, '\\');
+        $path = Path::canonicalize($value);
 
-        for ($i = 0, $c = \count($chunks); $i < $c; ++$i) {
-            if ('.' === $chunks[$i]) {
-                ++$i;
-                continue;
-            }
-
-            // Reduce multiple slashes to one
-            if (0 === strncmp($chunks[$i], '/', 1)) {
-                $resolved[] = '/';
-                continue;
-            }
-
-            // Reduce multiple backslashes to one
-            if (0 === strncmp($chunks[$i], '\\', 1)) {
-                $resolved[] = '\\';
-                continue;
-            }
-
-            if ('..' === $chunks[$i]) {
-                ++$i;
-                array_pop($resolved);
-                array_pop($resolved);
-                continue;
-            }
-
-            $resolved[] = $chunks[$i];
-        }
-
-        return rtrim(implode('', $resolved), '\/');
+        return $windowsPath ? str_replace('/', '\\', $path) : $path;
```

Comment:
Why do we need a special windows path handling here?

Reply:
This is because - judging from the tests - a windows path like `C:\a\b\c` should keep it's slashes. 

I'm not 100% if this is needed. If not we could drop the custom canonicalization altogether.

Reply:
@contao/developers anyone an idea why we want windows paths with backslashes here?

Reply:
This line: 
https://github.com/contao/contao/blob/d2eb923c90ec9b26fc69579309b89a334933490c/core-bundle/tests/DependencyInjection/ConfigurationTest.php#L75

would then have to be:

```php
        $this->assertSame('C:/Temp/contao', $configuration['image']['target_dir']);
```

Reply:
@leofeyer what do you think about removing the special windows handling here?

Reply:
If it works, I am all for it.

Reply:
Adjusted in 24145736b7f726f9ccaf41dfa8f51f0b1af953e0.